File: Problems/python/boj_로프.py
#https://www.acmicpc.net/problem/2217
#Greedy, Sort

k = int(input())
w = [int(input()) for i in range(k)]
w.sort(reverse=True)

mw = [] 
for i in range(k):
    mw.append((i+1)*w[i])

mw.sort()
print(mw[-1])

'''
x>y 일때 x*k<y*(k+1) 이 가능.
따라서 mw는 끝까지 다 구한후 최대값 찾음.
'''

File: Problems/cpp/boj_로프.cpp
//https://www.acmicpc.net/problem/2217
//Greedy, Sort

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    vector<int> r;
    int n, tmp, w=0;
    cin >> n;
    for (int i=0; i<n; i++){
        cin >> tmp;
        r.push_back(tmp);
    }

    sort(r.rbegin(), r.rend());

    for (int i=0; i<n; i++){
        int weight = (i+1) * r[i];
        if (w<weight)
            w = weight;
    }
    
    cout << w;
}